Gems/Multiplayer/Code/Include/Multiplayer/NetworkTime/RewindableObject.inl

```diff
@@ -66,6 +66,12 @@ namespace Multiplayer
         return GetValueForTime(GetCurrentTimeForProperty());
     }
 
+    template <typename BASE_TYPE, AZStd::size_t REWIND_SIZE>
+    inline const BASE_TYPE& RewindableObject<BASE_TYPE, REWIND_SIZE>::GetPrevious() const
+    {
+        return GetValueForTime(GetCurrentTimeForProperty() - HostFrameId(1));
```

Comment:
Ooph, careful.. put that -1 inside the GetCurrentTimeForProperty() instead because this will do really bad things on the autonomous client. For more clarity, when we perform the scoped rewind on the server, we *don't* rewind the predictable properties of the autonomous player because we're just about to tick them forward. Basically, 'Don't rewind the shooter'.

Reply:
That would just subtract a millisecond vs getting us the previous value wouldn't it? I'll need to look at that player case. Granted this also needs more testing. I'm hoping to pull your changes to MPSample in first though.

Reply:
Did a more in depth dive here and this should be fine. This will just grab the value for one frame behind perceived current. It's isolated to sync rewind usages.

Reply:
Nope, you've missed something _really_ critical..  That GetValueForTime call eventually drills down into this function:
```
    HostFrameId NetworkTime::GetHostFrameIdForRewindingConnection(AzNetworking::ConnectionId rewindConnectionId) const
    {
        return (IsTimeRewound() && (rewindConnectionId == m_rewindingConnectionId)) ? m_unalteredFrameId : m_hostFrameId;
    }
```

Reply:
Actually.. on further thought, maybe we could just put an assert inside this new function.. if anyone calls GetPrevious on a predictive property for the input processing player; we could just make that an error condition.

Reply:
Right, that part was intentional since it's how Get operates. GetPrevious is always the value in history one frame behind Get, even in rewind contexts. That being said, an assert makes sense since there are some contexts where it wouldn't make sense to call. It's really only intended for SyncRewind after all.

Reply:
I updated the function description for now to clarify the usage. I'll revisit the Assert once I can test edge cases more thoroughly with weapons in place.